pkgs/applications/misc/resp-app/default.nix

```diff
@@ -0,0 +1,116 @@
+{ stdenv
+, mkDerivation
+, lib
+, fetchFromGitHub
+, brotli
+, lz4
+, pyotherside
+, python3
+, python3Packages
+, qtbase
+, qtcharts
+, qtdeclarative
+, qtgraphicaleffects
+, qtimageformats
+, qmake
+, qtquick1
+, qtquickcontrols2
+, qtsvg
+, qttools
+, snappy
+, wrapQtAppsHook
+, zstd
+}:
+
+let
+  redis-rdb-tools = with python3Packages; buildPythonPackage rec {
+    pname = "redis-rdb-tools";
+    version = "resp_rdbtools-0.1.15";
+
+    src = fetchFromGitHub {
+      owner = "uglide";
+      repo = "redis-rdb-tools";
+      rev = version;
+      sha256 = "sha256-uoDNJ0Yzyf/4VM1n6e2rsy7/IdCJNdnD/HcqTRCkjwY=";
+    };
+
+    propagatedBuildInputs = [
+      redis
+    ];
+
+    doCheck = false;
+  };
+in
+mkDerivation rec {
+  pname = "RESP.app";
+  version = "2022.5";
+
+  src = fetchFromGitHub {
+    owner = "RedisInsight";
+    repo = "RedisDesktopManager";
+    fetchSubmodules = true;
+    rev = version;
+    sha256 = "sha256-5eI3J2RsYE5Ejb1r8YkgzmGX2FyaCLFD0lc10J+fOT4=";
+  };
+
+  nativeBuildInputs = [
+    python3Packages.wrapPython
+    qmake
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    brotli lz4 pyotherside python3 qtbase qtdeclarative qtgraphicaleffects
+    qtimageformats qtquick1 qtquickcontrols2 qtsvg qttools qtcharts snappy zstd
+  ] ++ pythonPath;
+
+
+  pythonPath = with python3Packages; [
+    bitstring
+    cbor
+    msgpack
+    phpserialize
+    redis-rdb-tools
+    python-lzf
+  ];
+
+  patchPhase = ''
+    substituteInPlace src/resp.pro \
+      --replace 'which ccache' "false"
+    substituteInPlace src/resp.pro \
+      --replace 'target.files = $$DESTDIR/resp' "${placeholder "src"}/bin/linux/release/resp"
+    substituteInPlace src/resp.pro \
+      --replace '/opt/resp_app' "${placeholder "out"}"
+    substituteInPlace src/resp.pro \
+      --replace 'target.path = $$LINUX_INSTALL_PATH' 'target.path = $$LINUX_INSTALL_PATH/bin'
+    substituteInPlace src/resp.pro \
+      --replace '/usr/' "$out/"
+  '';
+
+  qmakeFlags = [
+    "SYSTEM_LZ4=1"
+    "SYSTEM_ZSTD=1"
+    "SYSTEM_SNAPPY=1"
+    "SYSTEM_BROTLI=1"
+    "VERSION=${version}"
+    "src/resp.pro"
+  ];
+
+  dontWrapQtApps = true;
+
+  postInstall = ''
+    buildPythonPath "$out $pythonPath"
+    wrapProgram $out/bin/resp \
```

Comment:
Isn't this double wrapping?

Reply:
Without this double wrapping, I couldn't get our packaged `pyotherside` into the wrapper and saw this error:
```
"PyOtherSide error: Traceback (most recent call last):\n\n  File \"qrc:/python//formatters/__init__.py\", line 1, in <module>\n    from .binary import BinaryFormatter\n\n  File \"qrc:/python//formatters/binary.py\", line 1, in <module>\n    import bitstring\n\nModuleNotFoundError: No module named 'bitstring'\n"
```

Reply:
Got it to work using
```
  preFixup = ''
    buildPythonPath "$pythonPath"
    qtWrapperArgs+=(--prefix PYTHONPATH : "$program_PYTHONPATH")
  '';
```